std-bits/table/src/main/java/org/enso/table/format/xlsx/Range.java

```diff
@@ -0,0 +1,241 @@
+  }
+
+  /**
+   * Convert an Excel Column Name (e.g. DCR) into the index (1-based)
+   *
+   * @param column name
+   * @return Column index (A=1 ...)
+   */
+  public static int parseA1Column(CharSequence column) throws IllegalArgumentException {
+    ParsedInteger parsed = parseColumn(column);
+    if (parsed.index != column.length() || parsed.value == 0) {
+      throw new IllegalArgumentException(column + " is not a valid Excel Column Name.");
+    }
+
+    return parsed.value;
+  }
+
+  private static class ParsedInteger {
+    public final int index;
+    public final int value;
+
+    public ParsedInteger(int index, int value) {
+      this.index = index;
+      this.value = value;
+    }
+  }
+
+  private static ParsedInteger parseInteger(CharSequence address, int index) {
+    int endIndex = index;
+    while (endIndex < address.length() && isDigit(address.charAt(endIndex))) {
+      endIndex++;
+    }
+    return new ParsedInteger(
+        endIndex, endIndex == index ? 0 : Integer.parseInt(address, index, endIndex, 10));
+  }
+
+  private static ParsedInteger parseColumn(CharSequence column) {
+    int col = 0;
+
+    int index = skipDollar(column, 0);
+
+    while (index < column.length() && isLetter(column.charAt(index))) {
+      col = 26 * col + (column.charAt(index) - 'A' + 1);
+      index++;
+    }
+
+    return new ParsedInteger(index, col);
+  }
+
+  private final String sheetName;
+  private final int leftColumn;
+  private final int rightColumn;
+  private final int topRow;
+  private final int bottomRow;
+
+  public Range(String fullAddress) throws IllegalArgumentException {
+    String[] sheetAndRange = parseFullAddress(fullAddress);
+    this.sheetName = sheetAndRange[0].replaceAll("^'(.*)'$", "$1").replaceAll("''", "'");
+
+    int[] range = parseRange(sheetAndRange[1]);
+    this.leftColumn = range[1];
+    this.rightColumn = range[3];
+    this.topRow = range[0];
+    this.bottomRow = range[2];
+  }
+
+  public Range(String sheetName, int leftColumn, int topRow, int rightColumn, int bottomRow) {
+    this.sheetName = sheetName;
+    this.leftColumn = Math.min(leftColumn, rightColumn);
+    this.topRow = Math.min(bottomRow, topRow);
+    this.rightColumn = Math.max(leftColumn, rightColumn);
+    this.bottomRow = Math.max(bottomRow, topRow);
+  }
+
+  public String getSheetName() {
+    return sheetName;
+  }
+
+  public boolean isWholeRow() {
+    return leftColumn == 0;
+  }
+
+  public int getLeftColumn() {
+    return leftColumn;
+  }
+
+  public int getRightColumn() {
+    return rightColumn;
+  }
+
+  public boolean isWholeColumn() {
+    return topRow == 0;
+  }
+
+  public int getTopRow() {
+    return topRow;
+  }
+
+  public int getBottomRow() {
+    return bottomRow;
+  }
+
+  public String getAddress() {
+    String sheetNameEscaped = getSheetName();
+    if (sheetNameEscaped.contains(" ") || sheetNameEscaped.contains("'")) {
+      sheetNameEscaped = "'" + sheetNameEscaped.replace("'", "''") + "'";
+    }
```

Comment:
Feels like this and the dual in the constructor maybe could become standalone functions for (un)escaping the sheet name? I think it would be cleaner.